pandas/tools/merge.py

```diff
@@ -1411,6 +1411,10 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
             sample = objs[0]
         self.objs = objs
 
+        # Check for string axis parameter
+        if isinstance(axis, str):
+            axis = objs[0]._get_axis_number(axis)
```

Comment:
That would have been indeed have been useful, but it is not a class method .. (the attributes like `_AXIS_NAMES` are also only defined in the subclasses). And I don't think we have this logic already somewhere else (as in other methods you can just used the instance method)